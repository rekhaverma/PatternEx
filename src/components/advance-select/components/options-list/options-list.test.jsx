import React from 'react';
import OptionList from './options-list.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<OptionList />', () => {
  it('should match with snapshot', () => {
    const props = {
      options: [{
        id: 'string',
        value: 'string',
        isSelected: true
      }],
      onOptionClick: jest.fn(),
      dropDownHeader: 'Filter',
    };
    const wrapper = shallow(<OptionList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - autocomplete', () => {
    const props = {
      allowSearch: true,
      options: [{
        id: 'string',
        value: 'string',
        isSelected: true
      }],
      onOptionClick: jest.fn(),
      dropDownHeader: 'Filter',
    };
    const wrapper = shallow(<OptionList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
