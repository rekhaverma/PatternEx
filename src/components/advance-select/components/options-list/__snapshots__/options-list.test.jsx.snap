// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionList /> should match with snapshot - autocomplete 1`] = `
<div
  className="advance-select__option-list"
>
  <div
    className="advance-select__option-list-header"
  >
    Filter
  </div>
  <div
    className="advance-select__option-list-auto-complete"
  >
    <MatField
      errorMessage="default.search.string"
      hasError={false}
      label="default.search.string"
      name="autocomplete"
      onChange={[Function]}
      onErrorFound={[Function]}
      prefix="autocomplete"
      type="text"
      value=""
    />
  </div>
  <Scrollbars
    autoHeight={true}
    autoHeightMax={200}
    autoHeightMin={25}
    autoHide={false}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded={true}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <Option
      className="advance-select"
      id="string"
      isDisabled={false}
      isSelected={true}
      key="string"
      onOptionClick={[Function]}
      value="string"
    />
  </Scrollbars>
</div>
`;

exports[`<OptionList /> should match with snapshot 1`] = `
<div
  className="advance-select__option-list"
>
  <div
    className="advance-select__option-list-header"
  >
    Filter
  </div>
  <Scrollbars
    autoHeight={true}
    autoHeightMax={200}
    autoHeightMin={25}
    autoHide={false}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded={true}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <Option
      className="advance-select"
      id="string"
      isDisabled={false}
      isSelected={true}
      key="string"
      onOptionClick={[Function]}
      value="string"
    />
  </Scrollbars>
</div>
`;
